### ISSUE TITLE
[BUG] a11y: Toolbar buttons cannot be activated using keyboard

### ISSUE THREAD

**Issue by michaelweghorn** [bibletime/bibletime#357]
**Bug description**
Currently, it is not possible to activate toolbar items/buttons using the space key on the keyboard. (This - among others - negatively affects accessibility.)

**Steps to reproduce**
1. press the Tab key a few times until e.g. the "Display Settings" toolbar item is focused
2. press the space key on the keyboard

**Actual behaviour**
Nothing happens.

**Expected behavior**
The display settings menu should open, as is the case when left-clicking on the "Display Settings" toolbar item.

**System information:**
 - OS and OS version: Debian testing
 - BibleTime Version: built from current git master: 3.1.0dev+d757e8ebbb06186b73d1ecf50e3bcc1176105032
 - How BibleTime was installed: compiled from source code
 - (If compiled from source) Sword Version: 1.9.0+dfsg-4
 - (If compiled from source) Qt Version: 5.15.2

**Additional context**
My initial motivation for looking at BibleTime accessibility was a recent discussion about accessibility of Bible programs on the Debian accessibility mailing, thread starting at https://lists.debian.org/debian-accessibility/2021/11/msg00023.html .
If there's interest in improving accessibility in BibleTime, I'd generally be interested and willing to look into that topic a bit more and report issues or possibly even analyze and improve one or the other aspect. I'd be interested in hearing about your opinion on that.

**Comment by michaelweghorn** [comment]
I have analysed this a bit. The reason why this currently doesn't work is that the space key is used as a shortcut for the "Auto scroll pause" action. Keyboard activation works when that shortcut is removed, as I did for testing locally:

        --- a/src/frontend/bibletime_init.cpp
        +++ b/src/frontend/bibletime_init.cpp
        @@ -122,7 +122,7 @@ void BibleTime::insertKeyboardActions( BtActionCollection* const a ) {
         
             action = new QAction(a);
             action->setText(tr("Auto scroll pause"));
        -    action->setShortcut(QKeySequence(Qt::Key_Space));
        +    //action->setShortcut(QKeySequence(Qt::Key_Space));
             a->addAction("autoScrollPause", action);

Would it maybe be a possible solution to use a different shortcut for that instead? (Or somehow only have that shortcut in place while scrolling actually takes place?)

**Comment by jaakristioja** [comment]
I can confirm this issue. Using the space key as a default shortcut is indeed somewhat problematic, however I'm not sure what the best fix would be. Maybe scrolling should pause or stop whenever widget focus is transferred?

@gholmlund What do you think?

**Comment by gholmlund** [comment]
I leave the scroll pause action disabled unless the auto scroll is active.

**Comment by michaelweghorn** [comment]
Thanks a lot!

One scenario I ran into while testing:

* press Shift + Down to start autoscroll mode
* press pause to pause autoscroll
* press Tab a few times to navigate between toolbar buttons
* press space key

My personal expectation would have been that space activates the toolbar button, but what happens is that autoscroll starts again. Do you think it would be possible (and make sense) to stop autoscroll when moving focus e.g. as described above?
